### PR TITLE
feature: add docker and database setup, fix interface & init typescript

### PR DESCRIPTION
**Pull request by kleyow**

### CODE REVIEW

**Comment by kleyow** [comment]
This is basically a copy of auth-service with the prep work needed.

File: .gitignore [hunk 1]
```diff
@@ -2,58 +2,77 @@
...
-#Docs folder created for documentation
-docs
-
 # nyc test coverage
 .nyc_output
 
 # Grunt intermediate storage (http://gruntjs.com/creating-plugins#storing-task-files)
 .grunt
 
+# Bower dependency directory (https://bower.io/)
```
Comment (eoln) [inline .gitignore:26]: wow, I didn't notice we are using bower :-D

File: src/lib/db.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Raman Mangla <ramanmangla@google.com>
+ --------------
+ ******/
+
+import Knex from 'knex'
+import Config from '../shared/config'
+import ConsentDB from '../model/consent'
```
Comment (eoln) [inline src/lib/db.ts:32]: Do we need consentDB -> Shouldn't we only store FspId mapping only?

File: src/model/consent/consent.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
+import { NotFoundError } from '../errors'
```
Comment (eoln) [inline src/model/consent/consent.ts:1]: do we need consent model?

File: src/server/handlers/hello.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
+/*****
```
Comment (eoln) [inline src/server/handlers/hello.ts:1]: I think we can drop whole `hello` endpoint - it was implemented only as an example...

File: test/integration/model/consent.test.ts [hunk 5]
```diff
@@ -0,0 +1,176 @@
+import Knex from 'knex'
```
Comment (eoln) [inline test/integration/model/consent.test.ts:1]: do we need consent model & tests?

**Review by eoln — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (kleyow) [reply .gitignore]: Must of been a catchall sort of ignore in the auth-service.
done.

File: .gitignore [hunk 6]
```diff
@@ -2,58 +2,77 @@
...
+# Yarn Integrity file
+.yarn-integrity
+
+# dotenv environment variables file
+.env
+
+# next.js build output
+.next
 
-.npmrc
```
[the remark this answers is not in the document]
Comment (kleyow) [reply .gitignore]: done

File: .gitignore [hunk 7]
```diff
@@ -2,58 +2,77 @@
...
 # Coverage directory used by tools like istanbul
 coverage
 
-#Docs folder created for documentation
-docs
-
 # nyc test coverage
 .nyc_output
 
 # Grunt intermediate storage (http://gruntjs.com/creating-plugins#storing-task-files)
```
[the remark this answers is not in the document]
Comment (kleyow) [reply .gitignore]: done

File: .dockerignore [hunk 8]
```diff
@@ -0,0 +1,10 @@
+
+.dockerignore
+.eslintignore
+.git/
+.gitignore
+.circleci/
+
+Dockerfile
+
+node_modules/
```
[the remark this answers is not in the document]
Comment (kleyow) [reply .dockerignore]: done

File: README.md [hunk 9]
```diff
@@ -1 +1,60 @@
-# als-oracle-template
+# Auth Service (Work in Progress)
+[![Git Commit](https://img.shields.io/github/last-commit/mojaloop/als-consent-oracle.svg?style=flat)](https://github.com/mojaloop/als-consent-oracle/commits/master)
+[![Git Releases](https://img.shields.io/github/release/mojaloop/als-consent-oracle.svg?style=flat)](https://github.com/mojaloop/als-consent-oracle/releases)
+[![Npm Version](https://img.shields.io/npm/v/@mojaloop/als-consent-oracle.svg?style=flat)](https://www.npmjs.com/package/@mojaloop/als-consent-oracle)
+[![NPM Vulnerabilities](https://img.shields.io/snyk/vulnerabilities/npm/@mojaloop/als-consent-oracle.svg?style=flat)](https://www.npmjs.com/package/@mojaloop/als-consent-oracle)
+[![CircleCI](https://circleci.com/gh/mojaloop/als-consent-oracle.svg?style=svg)](https://circleci.com/gh/mojaloop/als-consent-oracle)
+
+Mojaloop central AuthZ + AuthN service. Currently for FIDO implementation in a Mojaloop switch.
```
[the remark this answers is not in the document]
Comment (kleyow) [reply README.md]: done.

File: README.md [hunk 10]
```diff
@@ -1 +1,60 @@
...
+
+## Overview
+
+- [Documentation](./docs/README.md)
+
+## Setup
+
+### Clone repo
+```bash
+git clone git@github.com:mojaloop/AuthService.git
```
[the remark this answers is not in the document]
Comment (kleyow) [reply README.md]: done.

File: CHANGELOG.md [hunk 11]
```diff
@@ -0,0 +1,133 @@
+# Changelog: [mojaloop/als-consent-oracle](https://github.com/mojaloop/als-consent-oracle)
```
[the remark this answers is not in the document]
Comment (kleyow) [reply CHANGELOG.md]: yep. I had all stuff fixed in the first PR and forgot to redo them again x - x.
done.

File: package.json [hunk 12]
```diff
@@ -1,49 +1,137 @@
...
-        "test:unit": "jest --testMatch '**/test/unit/**/*.test.js'",
-        "test:junit": "jest --reporters=default --reporters=jest-junit --testMatch '**/test/unit/**/*.test.js'",
-        "test:coverage": "jest --coverage --coverageThreshold='{}' --testMatch '**/test/unit/**/*.test.js'",
-        "test:coverage-check": "jest --coverage --testMatch '**/test/unit/**/*.test.js'",
-        "lint": "eslint .",
-        "audit:resolve": "SHELL=sh resolve-audit --production",
-        "audit:check": "SHELL=sh check-audit --production",
-        "build:openapi": "swagger-cli bundle -t yaml -o src/interface/openapi.yaml src/interface/openapi_template.yaml && swagger-cli validate src/interface/openapi.yaml"
+  "name": "@mojaloop/als-consent-oracle",
+  "version": "11.2.8",
```
[the remark this answers is not in the document]
Comment (kleyow) [reply package.json]: nope. change to 0.0.1 for now

File: package.json [hunk 13]
```diff
@@ -1,49 +1,137 @@
...
+  "description": "PISP authorization service",
```
[the remark this answers is not in the document]
Comment (kleyow) [reply package.json]: done

File: README.md [hunk 14]
```diff
@@ -1 +1,60 @@
-# als-oracle-template
+# Auth Service (Work in Progress)
```
[the remark this answers is not in the document]
Comment (kleyow) [reply README.md]: done

File: README.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (kleyow) [reply README.md]: done

File: package.json [hunk 15]
```diff
@@ -1,49 +1,137 @@
...
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/mojaloop/als-consent-oracle.git"
+  },
+  "keywords": [
+    "metrics",
+    "central",
+    "services"
+  ],
+  "author": "lewisdaly",
```
[the remark this answers is not in the document]
Comment (kleyow) [reply package.json]: done

File: docs/ErDiagram.puml [hunk 16]
```diff
@@ -0,0 +1,31 @@
+@startuml
```
[the remark this answers is not in the document]
Comment (kleyow) [reply docs/ErDiagram.puml]: Nope removed contents for now. I do want to add a diagram but that can wait for another pass.

File: src/lib/db.ts (hunk 2 above)
Comment (kleyow) [reply src/lib/db.ts:32]: Yes all the consent stuff relates to the mapping. ConsentDb will be that mapping. 
Like so 
```
  {
    id: 'ab7f9872-d9fe-4aee-a540-fc12f9aad8aa',
    fspId: 'dfspa'
  }
 ```
I thought the name was intutiative enough, maybe it's not, maybe `ConsentFspMapping`?

File: src/server/handlers/hello.ts (hunk 4 above)
Comment (kleyow) [reply src/server/handlers/hello.ts]: done

File: src/model/consent/consent.ts (hunk 3 above)
Comment (kleyow) [reply src/model/consent/consent.ts:1]: relevant to https://github.com/mojaloop/als-consent-oracle/pull/3#discussion_r560924716

File: test/integration/model/consent.test.ts (hunk 5 above)
Comment (kleyow) [reply test/integration/model/consent.test.ts:1]: relevant to https://github.com/mojaloop/als-consent-oracle/pull/3#discussion_r560924716

File: .gitignore (hunk 6 above)
[the remark this answers is not in the document]
Comment (eoln) [reply .gitignore]: +1

File: .gitignore (hunk 7 above)
[the remark this answers is not in the document]
Comment (eoln) [reply .gitignore]: +1

File: docs/ErDiagram.puml (hunk 16 above)
[the remark this answers is not in the document]
Comment (eoln) [reply docs/ErDiagram.puml]: +1

File: README.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (eoln) [reply README.md]: +1

File: src/lib/db.ts (hunk 2 above)
Comment (eoln) [reply src/lib/db.ts:32]: +1

**Review by eoln — APPROVED** [review]
lgtm

**Pull request merged**